### PR TITLE
ci: enable lightning on TPU

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

add parametrization for running TPU testing for the lightning mono package

Fixes #\<issue_number>

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/workflows/tpu-tests.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Test PyTorch - TPU
 on:
   push:
     branches: [master, "release/*"]
-  pull_request_target:
+  pull_request:  # fixme: pull_request_target
```
Comment (akihironitta) [inline .github/workflows/tpu-tests.yml:6]: Should this be reverted before merging this PR?
```suggestion
  pull_request:  # fixme: pull_request_target
```

**Review by akihironitta — APPROVED** [review]

**Comment by Borda** [comment]
> We need to decrease the TPU CI load, and adding one more parametrization will make things worse.

so you suggest rather don't know if it works?

**Comment by carmocca** [comment]
Yes, exactly :wink:

Now, why?
- TPU capacity is low.
- We already test `lightining` with the CPU and CUDA workflows
- We effectively already don't know if anything works because of the low capacity.
- Developments happen with `lightning` (not the standalone versions) so we'd catch issues when running on TPUs as we develop.

Alternatively, we could test `lightning.pytorch` and `lightning.fabric`, but my last point suggest that the current approach is better.

**Comment by Borda** [comment]
> * We already test `lightining` with the CPU and CUDA workflows

sounds like we do not need to test the motor-bike when we are already testing a bicycle just because both have two wheels... :chipmunk: 

> * TPU capacity is low.

then leave it as standalone extension when the load will be limited

> * We effectively already don't know if anything works because of the low capacity.

it is not a reason to don't test it in future...

> * Developments happen with `lightning` (not the standalone versions) so we'd catch issues when running on TPUs as we develop.

as in some of your PR you limit the testing/triggering events making it the same case standalone... so when you edit TPU code it would be in the other place and if general utils (you already suggested skipping TPU testing so no change in fact) and we have compatibility repo for this case...

**Comment by Borda** [comment]
closing as @carmocca does not think testing `lightning` is needed, lets's just whole all will work somehow... :chipmunk:

**Pull request closed without merging**